Mobile/ShopLens/Droid/SpeechActivity.cs

```diff
@@ -4,29 +4,40 @@
 using Android.OS;
 using Android.Widget;
 using Android.Speech;
+using Android.Speech.Tts;
+using Java.Util;
+using Android.Runtime;
 
 namespace ShopLens.Droid
 {
     [Activity(Label = "SpeechActivity")]
-    public class SpeechActivity : Activity
+    public class SpeechActivity : Activity, TextToSpeech.IOnInitListener
     {
         bool isRecording;
         readonly int VOICE = 10;
         TextView textBox;
         Button recButton;
+        Button recAndVoiceButton;
+
+        TextToSpeech textVoicer;
```

Comment:
I tried to make it so that SpeechActivity wouldn't need to have a textVoicer: I made a SpeakOut(string message) method in TextVoicerActivity and gave SpeechActivity an instance of TextVoicerActivity, but there was no way for me to call the method OnInit within the SpeakOut(string message). So, I ended up adding a textVoicer to this class.